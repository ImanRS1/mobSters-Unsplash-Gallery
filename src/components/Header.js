import React from "react";
import styled from "styled-components";

const Header = () => {
  return (
    <HeaderS>
      <Logo>mobSters Gallery</Logo>
    </HeaderS>
  );
};

const HeaderS = styled.header`
  width: 100vw;
  height: 5rem;
  position: fixed;
  top: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #fcf6f5ff;
  border-bottom: 4px solid #a13941ff;
  z-index: 5;
`;

const Logo = styled.h1`
  margin: 0;
  text-align: center;
`;

export default Header;
