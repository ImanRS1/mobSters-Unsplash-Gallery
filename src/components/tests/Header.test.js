import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from '../Header';

describe('Basic Header test', () => {
  const div = document.createElement('div');

  test('Renders', () => {
    const input = 'carts';
    ReactDOM.render(<BrowserRouter><Header input={input}/></BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  test('Header snapshot', () => {
    const input = 'carts';
    const HeaderComponent = renderer.create(<BrowserRouter><Header input={input}/></BrowserRouter>).toJSON();
    expect(HeaderComponent).toMatchSnapshot();
  })
});