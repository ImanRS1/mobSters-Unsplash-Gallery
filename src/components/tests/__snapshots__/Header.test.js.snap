// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Header test Header snapshot 1`] = `
<header
  className="sc-bdvvtL hHHJBI"
>
  <h1
    className="sc-gsDKAQ dTOPoC"
  >
    mobSters Gallery
  </h1>
  <nav>
    <ul>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        href="/search/carts"
        onClick={[Function]}
      >
        Search
      </a>
      <a
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </ul>
  </nav>
</header>
`;
